runtime/src/bank.rs

```diff
@@ -1304,9 +1319,12 @@ impl Bank {
         }
         // if I'm the first Bank in an epoch, count, claim, disburse rewards from Inflation
 
-        // calculated as: prev_slot / (slots / year)
-        let slot_in_year =
-            (self.epoch_schedule.get_last_slot_in_epoch(prev_epoch)) as f64 / self.slots_per_year;
+        // calculated as: num_slots / (slots / year)
+        let num_slots = self
+            .epoch_schedule
+            .get_last_slot_in_epoch(prev_epoch)
+            .saturating_sub(self.get_inflation_start_slot());
```

Comment:
hmm, in this code, I think we don't pay rewards for the activation epoch...

I think the math should be like this?: `self.epoch_schedule.get_last_slot_in_epoch(prev_epoch) - self.epoch_schedule.get_last_slot_in_epoch(activation_epoch.saturating_sub(1))`

Anyway, this is good indicate we need to write a good unit test for this particular calculation by extracting this first...

such a test should check that `num_slots` is rougthly `320000`.

Reply:
also, we should be careful for genesis case...